src/main/java/net/dries007/tfc/objects/items/metal/ItemMetalBucket.java

```diff
@@ -0,0 +1,190 @@
+                            else if (fluid.getBlock() instanceof BlockLiquid)
+                            {
+                                // Vanilla water and lava, and hopefully nothing else
+                                try
+                                {
+                                    BlockLiquid flowingBlock = BlockStaticLiquid.getFlowingBlock(fluid.getBlock().getDefaultState().getMaterial());
+                                    worldIn.setBlockState(pos, flowingBlock.getDefaultState());
+                                }
+                                catch (IllegalArgumentException e) { /* Just skip */ }
+                            }
+                        }
+                        worldIn.playSound(null, pos, SoundEvents.ITEM_BUCKET_EMPTY, SoundCategory.PLAYERS, 1.0F, 1.0F);
+                        return ActionResult.newResult(EnumActionResult.SUCCESS, stack);
+                    }
+                }
+            }
+        }
+        return ActionResult.newResult(EnumActionResult.PASS, stack);
+    }
+
+    @Override
+    @Nonnull
+    public String getItemStackDisplayName(@Nonnull ItemStack stack)
+    {
+        IFluidHandler bucketCap = stack.getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY, null);
+        if (bucketCap != null)
+        {
+            FluidStack fluidStack = bucketCap.drain(CAPACITY, false);
+            if (fluidStack != null)
+            {
+                String fluidname = fluidStack.getLocalizedName();
+                return new TextComponentTranslation("item.tfc.wooden_bucket.filled", fluidname).getFormattedText();
+            }
+        }
+        return super.getItemStackDisplayName(stack);
+    }
+
+    @Override
+    public void getSubItems(CreativeTabs tab, NonNullList<ItemStack> items)
+    {
+        if (isInCreativeTab(tab))
+        {
+            items.add(new ItemStack(this));
+            for (Fluid fluid : getValidFluids())
+            {
+                ItemStack stack = new ItemStack(this);
+                IFluidHandlerItem cap = stack.getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY, null);
+                if (cap != null)
+                {
+                    cap.fill(new FluidStack(fluid, CAPACITY), true);
+                }
+                items.add(stack);
+            }
+        }
+    }
+
+    public Set<Fluid> getValidFluids () {
+        if (metal.equals(BLUE_STEEL)) {
+            Set<Fluid> blueFluids = FluidsTFC.getAllMetalFluids().stream().map(FluidWrapper::get).collect(Collectors.toSet());
+            blueFluids.add(LAVA);
+            return blueFluids;
+        } else if (metal.equals(RED_STEEL)) {
+            return FluidsTFC.getAllWrappers().stream().filter(x -> x.get(DrinkableProperty.DRINKABLE) != null).map(FluidWrapper::get).collect(Collectors.toSet());
+        }
+        else //noinspection unchecked
+        return (Set<Fluid>) Collections.EMPTY_SET; // No other metal buckets implemented
+    }
```

Comment:
This by design is hardcoded and not extensible. It should implement a similar condition to how the ceramic jug checks for fluids by using a config-supplied list of fluid names. Two config entries, one for blue steel, one for red steel. The current code is fine for default values.

Reply:
The ceramic jug uses exactly this same code as the red steel bucket, all drinkable fluids. The wooden bucket uses a manually configured fluid list. Which makes sense? Classic would have just fresh water for the red steel bucket, and just Lava for blue steel bucket. Do we want to leave these open as currently coded, make them manually configured lists, or mimic the classic behavior of just fresh water/lava?

Reply:
Whoops, I meant to say wooden bucket. Yes, they should both be config based fluid lists. Default to lava / water only.

Reply:
Done.